### PR TITLE
Adding more info on writing adaptive apps

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixes https://github.com/flutter/website/issues/6141

### CODE REVIEW

File: src/ui/layout/responsive/building-adaptive-apps.md [hunk 1]
```diff
@@ -397,6 +403,16 @@ return Padding(
 );
 ```
 
+Use `Theme.of(context).platform` for theming and
+design choices, like what kind of switches to show
+and general Cupertino/Material adaptions.
+It also helps an app work well on mobile web.
```
Comment (parlough) [inline src/ui/layout/responsive/building-adaptive-apps.md:409]: From reading, it's not clear to me how this helps an app work well on mobile web. Consider elaborating or dropping the sentence.

File: src/ui/layout/responsive/building-adaptive-apps.md [hunk 2]
```diff
@@ -397,6 +403,16 @@ return Padding(
...
+
+Use `defaultTargetPlatform` if you're implementing
+platform logic in flutter/widgets, or are otherwise
```
Comment (parlough) [inline src/ui/layout/responsive/building-adaptive-apps.md:412]: What is "flutter/widgets" here? It sounds like it's referring to the implementation of the [`flutter/widgets`](https://api.flutter.dev/flutter/widgets/widgets-library.html) library. I think we should drop the mention if so.

Comment (parlough) [reply src/ui/layout/responsive/building-adaptive-apps.md:412]: Hmm, taking a look at this whole paragraph, I think it's specific to implementers of Flutter as others should generally use `Theme.of(context).platform`.

File: src/ui/layout/responsive/building-adaptive-apps.md [hunk 3]
```diff
@@ -343,10 +341,18 @@ bool get isDesktopDeviceOrWeb => kIsWeb || isDesktopDevice;
...
-supported on the web target. As a result, this code checks
+supported on the web target. As a result, the above code checks
 for web first, and because of short-circuiting,
 Dart never calls `Platform` on web targets.
 
+Use `Platform`/`kIsWeb` when the logic absolutely <i>must</i>
+run for a given platform. For example,
+talking to a plugin that only works on iOS,
+or displaying a widget that only conforms to
+Play Store policy and not App Store.
```
Comment (parlough) [inline src/ui/layout/responsive/building-adaptive-apps.md:352]: Should this be something like?

```suggestion
Play Store policy and not the App Store's.
```

**Review by parlough — APPROVED** [review]
Thanks @sfshaza2! A few questions and suggestions:

File: src/ui/layout/responsive/building-adaptive-apps.md (hunk 3 above)
Comment (sfshaza2) [reply src/ui/layout/responsive/building-adaptive-apps.md]: Yes, it's like my English has taken vacation.

**Pull request merged**